.github/workflows/ngen_integration.yaml

```
@@ -0,0 +1,107 @@
+  # TODO: add something later that can check the runners for hyperthreading
+  LINUX_NUM_PROC_CORES: 2
+  MACOS_NUM_PROC_CORES: 3
+  ASAN_OPTIONS: detect_leaks=false
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Run general unit tests in linux environment
+  test_surfacebmi_plus_cfe:
+    # The type of runner that the job will run on
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: checkout actions in another repo
+        uses: actions/checkout@v3
+        with:
+          repository: noaa-owp/ngen
+
+      # Build some necessary submodules
+      - name: Build ISO C Fortran BMI
+        id: submod_build_2
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/iso_c_fortran_bmi/"
+
+      - name: Build PET
+        id: submod_build_5
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/evapotranspiration/evapotranspiration"
+          targets: "petbmi"
+
+      - name: Build SLoTH
+        id: submod_build_6
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/sloth/"
+          targets: "slothmodel"
+
+      # Build Ngen and save some files
+      - name: Build Ngen
+        id: ngen_id1
+        uses: ./.github/actions/ngen-build
+        with:
+          targets: "ngen"
+          build-cores: ${{ env.LINUX_NUM_PROC_CORES }}
+          #is this required for this test?
+          bmi_c: 'ON'
+          bmi_fortran: 'OFF'
+        timeout-minutes: 15
+
+      - name: Cleanup Ngen Build
+        run: |
+          # Move ngen build artifacts to temp directory
+          mv ${{ steps.ngen_id1.outputs.build-dir }} ${{runner.temp}}/ngen-build/
+          mv .github ${{runner.temp}}/.github
+          mv extern ${{runner.temp}}/extern
+          mv data ${{runner.temp}}/data
+
+      - name: Run surfacebmi plus cfebmi
+        run: |
+          echo "end of ngen_integration testing"
+
+      # Checkout and build CFE
+      - name: Checkout the commit
+        uses: actions/checkout@v4
+
+      - name: Build CFE Library for Ngen
+        run: |
+          cmake -B cmake_build -S . -DNGEN=ON
+          make -C cmake_build
+
+          # Move ngen build and other files to appropriate directory to run ngen
+          mv ${{runner.temp}}/ngen-build ./ngen-build
+          mv ${{runner.temp}}/extern/sloth extern
+          cp -r ${{runner.temp}}/extern/evapotranspiration extern
+          mv ${{runner.temp}}/data data
+
+          # Run ngen with CFE with PET
+          inputfile='realizations/ngen_realization_cfe_pet.json'
```

```suggestion
          inputfile='realizations/ngen_realization_cfe_pet_ngenCI.json'
```
It seems that this should match the file that's actually added

---

That's probably why the test failed.